### ISSUE TITLE
Warning: Could not connect to appstore: cURL error 28: Connection timed out after 60001 milliseconds

### ISSUE THREAD

**Issue by pieter-groeneweg** [nextcloud/server#23334]
this is second day in a row, It seems the appstor is unavailable>>

my setup can see there is an update available 19.0.4 but is not able to see the appstore. no checks.

I had no issues when I updated from 19.0.2 to 19.0.3 then al was ok.

full error message:
```
[appstoreFetcher] Warning: Could not connect to appstore: cURL error 28: Connection timed out after 60001 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)

GET /ocs/v2.php/apps/updatenotification/api/v1/applist/19.0.4.2
from xxx.xxx.xxx.xxx by uuuuuu at 2020-10-10T17:09:27+02:00
```

**Comment by nursoda** [comment]
This affects all versions using the app store including NC 20. In my opinion, this is a design flaw. Nextcloud settings/apps must stay functional no matter whether apps.nextcloud.com is available, slow or unavailable – because one not only updates apps there, but also may want to deactivate one or restrict an app to a certain group. If the app store is not responsive (enough), settings/apps should report (after less than a minute) that the update check is temporarily unavaliable.

@pieter-groeneweg Maybe you want to rephrase your title accordingly.

**Comment by colinscroggins** [comment]
+1 I agree with  @nursoda! This needs to be refactored such that local app management remains available and performant, even if the remote app store is down. Simple language notification that the app store is down would be nice, too.

**Comment by kesselb** [comment]
Closing in favour of https://github.com/nextcloud/server/issues/14926. Pull requests are always welcome ;)

**Comment by founderio** [comment]
@kesselb is there a status page for outages like this? Something like https://www.githubstatus.com/ which can be linked to in error messages so people don't accidentally DDoS your appstore via F5...

**Comment by kesselb** [comment]
I'm not aware of anyone.

**Comment by pieter-groeneweg** [comment]
Without some lay men terms explanation..I can't make any chocolate of all...... is it up or is it down?

**Comment by kesselb** [comment]
It seems there are some issues with a critical server. There is nothing you can do. Reschedule the update to next week.

**Comment by pieter-groeneweg** [comment]
@kesselb, That is all I needed to know.. Tx.

**Comment by founderio** [comment]
@kesselb Are you in a position to get people to think about creating such a status page or is this something that should be reported in an issue tracker somewhere? Not sure if this would fit nextcloud/server repo since it is about nextcloud.com, not the app itself..

**Comment by kesselb** [comment]
I'm not. 

https://github.com/nextcloud/nextcloud.com is the website repository.

**Comment by basvandenbrink** [comment]
Agree with @nursoda. The design flaw is also addressed in #10499.
